packages/ag-charts-enterprise/src/features/annotations/fibonacci-retracement-trend-based/fibonacciRetracementTrendBasedScene.ts

```diff
@@ -0,0 +1,331 @@
+import { _ModuleSupport } from 'ag-charts-community';
+
+import type { PointProperties } from '../annotationProperties';
+import type { AnnotationContext } from '../annotationTypes';
+import { AnnotationScene } from '../scenes/annotationScene';
+import { CollidableLine } from '../scenes/collidableLineScene';
+import { FibonacciScene } from '../scenes/fibonacciScene';
+import { DivariantHandle } from '../scenes/handle';
+import type { StartEndHandle } from '../scenes/startEndScene';
+import { getDragStartState, snapToAngle, translate } from '../utils/coords';
+import { createFibonacciRangesData, getFibonacciCoords } from '../utils/fibonacci';
+import { validateDatumPoint } from '../utils/validation';
+import { convertLine, convertPoint, invertCoords } from '../utils/values';
+import type { FibonacciRetracementTrendBasedProperties } from './fibonacciRetracementTrendBasedProperties';
+
+const { Vec2, Vec4 } = _ModuleSupport;
+
+type ActiveHandle = StartEndHandle | 'endRetracement';
+
+export class FibonacciRetracementTrendBasedScene extends FibonacciScene<FibonacciRetracementTrendBasedProperties> {
```

Comment:
Can some of these functions be moved into `FibonnaciScene`? Looks like `dragHandle()` is identical.

For others its just the `endRetracement` handle has been added in a few statements and calls. I'm not sure if there's any real benefit to making these ones more generic? At some point it becomes too convoluted when instead a bit of duplication is easier to work with.

But I'm interested in your thoughts.

Reply:
Discussed offline and decided to leave as is for now